front/components/agent_builder/capabilities/knowledge/transformations.ts

```diff
@@ -54,9 +84,30 @@ export function transformTreeToSelectionConfigurations(
       };
     }
 
-    // If it's not a full data source selection, mark as partial
-    if (!isFullDataSource && !configurations[dataSourceView.sId].isSelectAll) {
+    // If it's not a full data source selection, extract the selected nodes
+    if (!isFullDataSource) {
```

Comment:
my brain is not working to understand how everything works, I will try I have more energy but probably something for @johnoppenheimer to review